### PR TITLE
opt: add page

### PR DESCRIPTION
**Pull request by guilhermeleobas**
Hi,

I've added a page for the LLVM [`opt`](https://linux.die.net/man/1/opt) tool. 

----

- [x] The page (if new), does not already exist in the repo.

- [x] The page (if new), has been added to the correct platform folder:  
      `common/` if it's common to all platforms, `linux/` if it's Linux-specific, and so on.

- [x] The page has 8 or fewer examples.

- [x] The PR is appropriately titled:  
      `<command name>: add page` for new pages, or `<command name>: <description of changes>` for pages being edited.

- [x] The page follows the [contributing](https://github.com/tldr-pages/tldr/blob/master/CONTRIBUTING.md) guidelines.

### CODE REVIEW

File: pages/common/opt.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# opt
+
+> LLVM .bc -> .bc modular optimizer and analysis printer. `opt` takes LLVM source files as input, runs the specified optimizations or analyses on it, and then outputs the optimized file or the analysis results.
```
Comment (sbrl) [inline pages/common/opt.md]: This seems rather wordy. Can we make it more concise at all?

Also, what is meant by `.bc`?

File: pages/common/opt.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+- Run an optimization or an analysis in a bitcode file:
```
Comment (sbrl) [inline pages/common/opt.md]: Perhaps

```suggestion
- Run an optimization or analysis on a bitcode file:
```

....will read better?

Also, can `file.bc` be `path/to/file.bc` please?

File: pages/common/opt.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+`opt {{-passname}} {{file.bc}} -S -o {{file_opt.bc}}`
+
+- Print the Control Flow Graph of a function to a "dot" file:
```
Comment (sbrl) [inline pages/common/opt.md]: Perhaps `Output` would be a better term than `Print` here if it's going to a file and not a terminal?

Also, can `file.bc` be `path/to/file.bc` please?

File: pages/common/opt.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+`opt {{-dot-cfg}} -S {{file.bc}} -disable-output`
+
+- Optimize the program with "-O2" and write its output to another file:
```
Comment (sbrl) [inline pages/common/opt.md]: Perhaps we could specify what `-O2` actually does instead? How about this?

```suggestion
- Optimize the program at level 2 and output the result to another file:
```

Also, perhaps `file_opt.bc` can be `path/to/file_opt.bc` instead?

File: pages/common/opt.md (hunk 1 above)
Comment (guilhermeleobas) [reply pages/common/opt.md]: > This seems rather wordy. Can we make it more concise at all?

Ok. I will think on something else.

> Also, what is meant by .bc?

`.bc` is a bitcode file while `.ll` is a human-readable version of it.

**Comment by guilhermeleobas** [comment]
Hi @sbrl, thanks for the review. I've applied your suggestions.

File: pages/common/opt.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
+# opt
+
+> LLVM .bc -> .bc modular optimizer and analysis printer. `opt` takes LLVM source files as input, runs the specified optimizations or analyses on it, and then outputs the optimized file or the analysis results.
+
+- Run an optimization or analysis on a bitcode file:
+
+`opt {{-passname}} {{path/to/file.bc}} -S -o {{file_opt.bc}}`
```
Comment (agnivade) [inline pages/common/opt.md]: `-` of passname should be outside the tokens.

File: pages/common/opt.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+- Output the Control Flow Graph of a function to a "dot" file:
+
+`opt {{-dot-cfg}} -S {{path/to/file.bc}} -disable-output`
+
+- Optimize the program at level 2 and output the result to another file:
+
+`opt -O2 {{path/to/file.bc}} -S -o {{file_opt.bc}}`
```
Comment (agnivade) [inline pages/common/opt.md]: path/to/output_file.bc

File: pages/common/opt.md (hunk 1 above)
Comment (agnivade) [reply pages/common/opt.md]: Yea, this is a bit too long for a tldr page. How about - "A tool that takes LLVM source files and runs specified optimizations or analyses on them" ?

**Comment by guilhermeleobas** [comment]
Hi @agnivade and @sbrl, I've applied your suggestions but my changes aren't being reflected in this PR. Maybe it's because I naively deleted the fork of my other PR (#2499). 

I've tried to resync another fork with this PR [(ref)](https://github.com/isaacs/github/issues/168#issuecomment-374201226) but that didn't work. Maybe the best is if I open a new PR. What do you guys think?

Sorry for the trouble.

**Comment by sbrl** [comment]
Hrm. Each PR should end up on a new branch, @guilhermeleobas.

Personally, I use the Web UI for PRs & editing them for tldr-pages :P

**Comment by guilhermeleobas** [comment]
I am aware of that but look the header of this PR. It says the name of the base branch is **unknown branch** instead of **opt**:
image

**Comment by sbrl** [comment]
Ah, I see!

Yeah, I suspect that this PR and your previous one were from the same branch.

**Comment by guilhermeleobas** [comment]
I think I was using different branches. Because otherwise, my commits from the previous PR would appear in this one. Anyway, do you think the best approach is to open a new PR with the changes asked by @agnivade?

**Comment by agnivade** [comment]
Sure, let's do that.

**Pull request closed without merging**